src/functions-and-arrays.js

```diff
@@ -1,41 +1,151 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
-
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  }
+  return num2;
+}
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(arrWords) {
+  if (arrWords.length === 0) {
+    return null;
+  } else if (arrWords.length !== 0) {
+    let longestWordYet = '';
+    for (i = 0; i < arrWords.length; i++) {
+      if (arrWords[i].length > longestWordYet.length) {
+        longestWordYet = arrWords[i];
+      } else {
+        continue;
+      }
+    }
+    return longestWordYet;
+  }
+}
+
+/*
+if (arrWords.length === 0) {
+  return null
+}
+*^^^ another way: if (!arrWords.length)
+
+for (const currentWord of arrWords) {
+  if (logestWordYet.length < currentWord.length) {
+    LongestWordYet = currentWord
+  }
+}
 
+^^^ === :
+for (let i = 0; i < arrWords.length; i++) {
+  const currentWord
+}
 
+*/
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(nums) {
+  let sum = 0;
+  if (nums.length === 0) {
+    return 0;
+  } else if (nums.length !== 0) {
+    for (i = 0; i < nums.length; i++) {
+      sum += nums[i];
+    }
+  }
+  return sum;
+}
 
+/*
+function sumNumbers(numArray) {
 
+  let sum = 0;
 
-// Iteration #3.1 Bonus:
-function sum() {}
+for (const number of numArray) {
+  sum += number
+}
+return sum
+}
 
+*/
 
+// Iteration #3.1 Bonus:
+function sum(multiArray) {
+  let sum = 0;
+  for (let i = 0; i < multiArray.length; i++) {
+    if (multiArray[i] === true) {
+      sum += 1;
+    } else if (multiArray[i] === false) {
+      sum += 0;
+    } else if (typeof multiArray[i] === 'string') {
+      sum += multiArray[i].length;
+    } else if (typeof multiArray[i] === 'object') {
+      throw "Unsupported data type sir or ma'am";
+    } else {
+      sum += multiArray[i];
+    }
+  }
+  return sum;
+}
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
+function averageNumbers(nums) {
```

Comment:
You could have reused the sum function (which you built on line 50) here! Isn't reusing functions amazing? 😁